rfcs/2021-10-icons-in-core-DS.md

```diff
@@ -0,0 +1,102 @@
+# RFC: Standardizing Icons in the Core DS
+
+**Date**: October 01, 2021
+**Status**: `Review`
+
+## Problem
+
+The core DS does not have a consistent and easy way for consuming applications to import SVG icons. Additionally, because of the way some icons are used in DS components, it is difficult for consuming applications to change styles for an icon (like color, size, positioning, etc).
+
+### Background
+
+Currently, icons are stored in three main ways:
+
+1. Exported as an individual React component, but not in a centralized location
+2. Stored in the `design-system/images/` directory as .svg files
+3. Included in a component's markup unable to be re-used individually.
+
+The .svg files in the `images/` directory are referenced in the code in two main ways:
+
+1. In CSS as background images
+2. Using the library `inline-react-svg`, .svg files can be imported and used as a component (Note: this does not work with Typescript components)
+
+## Proposal
+
+Icons in the core DS should be moved to a centralized location. Each icon should be exported individually as a React component and updated with a standard size and consistent accessibilty features. This will provide more flexibility to consuming applications by allowing them to import individual SVG icons, adjust the styling on them with more ease, and create icon consistency among products.
+
+This ideal state can be achieved in a few phases:
+
+1. [Create icon components](#phase-1-create-icon-components)
+2. [Clean & improve markup](#phase-2-clean-and-improve-svg-markup)
+3. [Update icon references in components](#phase-3-update-component-icon-references)
+4. [Update icon references in CSS](#phase-4-update-css-icon-references)
+5. [Beyond](#beyond)
+
+### Phase 1: Create Icon Components
+
+In this phase:
+
+- Make `Icons` directory in the design-system component folder
+- Create a component for each icon in the `images` directory
+- Move the `ClearIcon` and `ClearIconAlternate` components into new directory
+- Move inlined SVG markup for `<` & `>` from Pagination to new directory
```

Comment:
While I was writing Pagination, I was thinking I could've just used 1 chevron icon and used CSS to rotate it the direction I wanted. Maybe we only need 1 of these chevron icons?

Reply:
I like that. We could honestly have a variation for up, down, left and right if needed. I'll update with clarification